FastSimulation/Calorimetry/src/KKCorrectionFactors.cc

```diff
@@ -0,0 +1,75 @@
+#include "FastSimulation/Calorimetry/interface/KKCorrectionFactors.h"
+
+KKCorrectionFactors::KKCorrectionFactors( const edm::ParameterSet& pset )
+{
+
+  interpolate_ = pset.exists("interpolate") && pset.getUntrackedParameter<bool>("interpolate");
+
+  // Get the filename and histogram name
+  std::string ecalScalesFileName_ = pset.getUntrackedParameter<std::string>("fileName");
+  std::string ecalScalesHistogramName_ = pset.getUntrackedParameter<std::string>("histogramName");
+
+  // Read histo
+  edm::FileInPath myDataFile( ecalScalesFileName_ ); // TODO: check if this is necessary
+  TFile * myFile = new TFile( myDataFile.fullPath().c_str(), "READ" );
+
+  gROOT->cd(); // create histogram in memory
+  auto obj = myFile->Get( ecalScalesHistogramName_.c_str() );
+
+  // Check if histogram exists in file
+  if(!obj) {
+    throw cms::Exception( "FileReadError" )
+      << "Histogram \"" << ecalScalesHistogramName_
+      << "\" not found in file \"" << ecalScalesFileName_
+      << "\", used for correcting the response of the ECAL in FastSim.\n";
+  }
+  h3_ = new TH3F( *((TH3F*)obj) );
+
+  // fill empty bins with 1.
+  for( int x=0; x<h3_->GetNbinsX()+2; x++ ) {
```

Comment:
This should be handled in the creation of the TH3.
E.g. what if you'd like to have some factors with value 0?